### DISCUSSION TITLE
Is bumping port-version strictly necessary if a change is cosmetic only?

### DISCUSSION THREAD

**Discussion by wrobelda** [microsoft/vcpkg#26865] [Q&A]
Take #26815 as an example: I merely updated the pre-compilation message which misinformed about a system package pre-requisite. Bumping port-version in this case will trigger not only this port's recompilation, which requires substantial time and energy, but also all of its upstream dependants. Obviously the recompilation is moot, as there's no inherent value added to newly recompiled binary.  

Is bumping a port-version strictly necessary in such cases? Is there a technical reason for it? It should be possible to simply update the existing version's SHA in `versions.json`, without adding a new one, shouldn't it?

I think it goes without saying that the current approach is _very_ anti-eco-friendly, considering how many users vcpkg has.

**Comment by wrobelda** [comment]
(feel free to convert this into a Discussion, as I believe it is more suitable there)

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. We have [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#port-version) about `port-version`, it says 
```
if you make a change to a port, without changing the underlying version of the library, you should increment this field by one (starting at 0, which is equivalent to no "port-version" field).
```
I will convert this issue to discussion, we can look forward to the opinions from other users.
